Trees/BinarySearchTree.cpp

```diff
@@ -0,0 +1,358 @@
+/*
+    Author:
+    Varun Jayathirtha (www.github.com/varunjm)
+
+    Binary Search Tree:
+    A graph data structure where each node has a unidirectional path to at most
+    two node, none of which are itself or its predecessors, i.e the graph is
+    acyclic. The first node is called the root node. Another important feature
+    of this tree is that each node has values lesser than or equal to itself in
+    the left subtree and values greater than or equal to itself in the right
+    subtree.
+*/
+
+#include "BinarySearchTree.hpp"
+#include <iostream>
+#include <climits>
+#include <stack>
+#define INF INT_MAX
+
```

Comment:
INT_MAX max is (sort of) a constant declared in the "climits" library. Why should i redeclare it as a constant? The only reason I have this line is to rename the constant to be more meaningful.